### DISCUSSION TITLE
Validating JWT when using Netlify Edge Function

### DISCUSSION THREAD

**Discussion by brewhousedigital** [pocketbase/pocketbase#937] [Q&A]
I'm doing some fancy behind-the-scenes updates using a Netlify Edge Function, and I wanted to know if it was possible to validate the JWT coming from the UI before I store data in PocketBase. My flow kind of looks like this:

1. Authenticated UI User
2. POST to Netlify Edge Function
3. Validate User's JWT **<-- Need Help Here**
4. Do some processing, send out mail, update another third party service, and save new data to PocketBase on the NodeJS side

Is this possible? I know I can get the ID out of the token but that's easy to modify on the client side. I want to make sure the JWT was signed correctly and not tampered with on the NodeJS side (Netlify Edge Function)

**Comment by ganigeorgiev** [comment]
Currently there is no dedicated method to validate a JWT, but you can achieve similar result by sending a `POST /api/collections/users/auth-refresh` call with the token in the `Authorization` header (for v0.7.x the route is `/api/users/refresh`). If the request succeed then the token is verified, otherwise - unverified (either expired, or invalid JWT).

If you are using the JS SDK, you can do something like this in v0.8.0-rc: 

```js
const pb = new PocketBase("http://127.0.0.1");

// store the token so that it can be send with the refresh request
pb.authStore.save("YOUR_TOKEN", null);

const isTokenVerified = pb.authStore.isValid && !!(await pb.collection('users').authRefresh().catch(() => false));
```

For v0.7.x is similar, but you have to pass some mock user model data in order to detect the Authorization schema:

```js
const pb = new PocketBase("http://127.0.0.1");

// store the token so that it can be send with the refresh request
pb.authStore.save("YOUR_TOKEN", { verified: false });

const isTokenVerified = pb.authStore.isValid && !!(await pb.users.refresh().catch(() => false));
```

`pb.authStore.isValid` is doing a loose check for the token `exp` claim to avoid sending unnecessary refresh since we know already that the token is invalid.

Optionally, to ensure that the token is for the expected user, you can get the result from the refresh call and check the returned user model id whether it matches with your expected one.
